test/unit/vdom/properties.js

```diff
@@ -117,4 +117,36 @@ describe('vdom/properties', () => {
       );
     });
   });
+
+  it('#876 - Apply properties and attributes in a polyfill environment', (done) => {
+    const Elem1 = define('x-test', {
+      render (elem) {
+        return h(Elem2, { foo: true, bar: true, baz: true });
+      }
+    });
+    const Elem2 = define('x-test', {
+      props: {
+        foo: {}
+      },
+      render(elem) {
+        expect(elem.foo).to.be.equal(true);
+        expect(elem.bar).to.be.equal(undefined);
+        expect(elem.baz).to.equal(true);
+        expect(elem.hasAttribute('foo')).to.equal(false);
+        expect(elem.getAttribute('bar')).to.equal('true');
```

Comment:
as far as i understand, vdom sets as prop if `elem.hasOwnProperty(name) === true` otherwise it sets as attr